<a href="https://colab.research.google.com/github/izzat-ai/learning-ai/blob/main/projects/movie.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
df = pd.read_csv("/content/Top_rated_movies.csv")
df.head()

,id,title,original_language,original_title,overview,release_date,popularity,vote_count,rating
0,13448,Angels & Demons,en,Angels & Demons,Harvard symbologist Robert Langdon is recruite...,2009-04-20,6.8861,7323,6.719
1,9603,Clueless,en,Clueless,"Shallow, rich and socially successful Cher is ...",1995-07-19,7.5190,4801,7.255
2,6557,27 Dresses,en,27 Dresses,Altruistic Jane finds herself facing her worst...,2008-01-10,5.7786,3421,6.453
3,1333100,Attack on Titan: THE LAST ATTACK,ja,劇場版「進撃の巨人」完結編 THE LAST ATTACK,A colossal-sized omnibus film bringing togethe...,2024-03-20,6.0979,205,8.700
4,578701,Those Who Wish Me Dead,en,Those Who Wish Me Dead,A young boy finds himself pursued by two assas...,2021-05-05,6.2045,1849,6.607


In [3]:
df2 = pd.read_csv("/content/movie_data.csv")
df2.head()

,id,title,original_language,overview,release_date,original_title,vote_average,vote_count
0,1339713,Obsession,en,"After breaking the mysterious ""One Wish Willow...",2026-05-13,Obsession,7.900,497
1,1304313,Lee Cronin's The Mummy,en,The young daughter of a journalist disappears ...,2026-04-15,Lee Cronin's The Mummy,8.056,1548
2,1433117,Kara,ta,"A thief tries to go straight, but when predato...",2026-04-30,கர,5.800,14
3,1380291,Tom Clancy's Jack Ryan: Ghost War,en,Jack Ryan is reluctantly pulled back into espi...,2026-05-20,Tom Clancy's Jack Ryan: Ghost War,7.105,457
4,1083381,Backrooms,en,A strange doorway appears in the basement of a...,2026-05-27,Backrooms,6.700,275


##EDA

In [4]:
df.columns

Index(['id', 'title', 'original_language', 'original_title', 'overview',
       'release_date', 'popularity', 'vote_count', 'rating'],
      dtype='object')

In [7]:
df.shape

(10000, 9)

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10000 entries, 0 to 9999
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   id                 10000 non-null  int64  
 1   title              10000 non-null  object 
 2   original_language  10000 non-null  object 
 3   original_title     10000 non-null  object 
 4   overview           9500 non-null   object 
 5   release_date       10000 non-null  object 
 6   popularity         10000 non-null  float64
 7   vote_count         10000 non-null  int64  
 8   rating             10000 non-null  float64
dtypes: float64(2), int64(2), object(5)
memory usage: 703.3+ KB


1. overview ustunida 500 ta ma'lumotlar yo'q
2. release_date ustuni object turida

In [9]:
df.describe()

,id,popularity,vote_count,rating
count,1.000000e+04,10000.000000,10000.000000,10000.000000
mean,6.165003e+05,6.433940,2765.800000,6.210350
std,6.230573e+05,0.547213,3236.140405,2.335112
min,9.750000e+02,5.473400,0.000000,0.000000
25%,9.526500e+03,6.087075,160.000000,5.960500
50%,5.127105e+05,6.466050,1383.500000,6.714500
75%,1.186288e+06,6.879800,4263.500000,7.487750
max,1.653208e+06,7.519000,11993.000000,8.700000


In [10]:
df.duplicated().sum()

np.int64(9980)

In [11]:
print(df.duplicated().sum())

9980


In [15]:
df.sample(5)

,id,title,original_language,original_title,overview,release_date,popularity,vote_count,rating
3044,578701,Those Who Wish Me Dead,en,Those Who Wish Me Dead,A young boy finds himself pursued by two assas...,2021-05-05,6.2045,1849,6.607
620,13448,Angels & Demons,en,Angels & Demons,Harvard symbologist Robert Langdon is recruite...,2009-04-20,6.8861,7323,6.719
9716,76203,12 Years a Slave,en,12 Years a Slave,"In the pre-Civil War United States, Solomon No...",2013-10-18,6.4827,11993,7.931
6201,9603,Clueless,en,Clueless,"Shallow, rich and socially successful Cher is ...",1995-07-19,7.5190,4801,7.255
6856,76203,12 Years a Slave,en,12 Years a Slave,"In the pre-Civil War United States, Solomon No...",2013-10-18,6.4827,11993,7.931


- Yuqorida duplikatlarni aniqladik , ular 9980 ta chiqdi ya'ni takrorlangan ma'lumotlar . Kaggle'dagi bu dataset API orqali yuklanayotganda yoki generatsiya qilinayotganda texnik xatolik tufayli aynan bir xil qatorlar qayta-qayta (yuzlab marta) nusxalanib, pastga qo'shilib ketgan
- Bu muammoni Data Cleaningda tozalab ketamiz

In [16]:
df['original_language'].value_counts()

,count
original_language,
en,7000
zh,1000
ja,500
tl,500
cn,500
hi,500
